$OpenBSD: patch-src_modules_loaders_loader_xpm_c,v 1.5 2009/04/30 10:09:49 simon Exp $

Chunk at line 246:
CVE-2008-5187, Buffer overflow in the xpm loader.
Patch from Debian's original bug report, #505714

--- src/modules/loaders/loader_xpm.c.orig	Tue Oct 21 04:32:51 2008
+++ src/modules/loaders/loader_xpm.c	Wed Apr 29 10:39:45 2009
@@ -253,8 +253,8 @@ load(ImlibImage * im, ImlibProgressFunction progress, 
                                  return 0;
                               }
                             ptr = im->data;
-                            end = ptr + (sizeof(DATA32) * w * h);
                             pixels = w * h;
+			    end = ptr + pixels;
                          }
                        else
                          {
